src/os/linux/local/mode/ntp.pm

```diff
@@ -209,39 +232,166 @@ sub get_ntp_modes {
             command => 'chronyc',
             command_options => '-n sources 2>&1',
             type => 'chronyc'
+        },
+        timedatectl => {
+            regexp => '',
+            command => '',
+            command_options => '',
+            type => 'timedatectl'
         }
     };
 
-    if ($self->{option_results}->{ntp_mode} eq 'ntpq') {
-        return [ $modes->{ntpq} ];
-    } elsif ($self->{option_results}->{ntp_mode} eq 'chronyc') {
-        return [ $modes->{chronyc} ];
+    # Returns the selected mode or all modes
+    return [ $modes->{$self->{option_results}->{ntp_mode}} ?
+                $modes->{$self->{option_results}->{ntp_mode}} :
+                ( $modes->{timedatectl}, $modes->{ntpq}, $modes->{chronyc} ) ];
```

Comment:
```suggestion
                ( $modes->{timedatectl}, $modes->{chronyc}, $modes->{ntpq} ) ];
```

Reply:
I suggest this because ntpq is older than chrony, which is older than timedatectl, and older should also mean rarer